hi-ml/src/health_ml/runner.py

```diff
@@ -0,0 +1,361 @@
+    """
+    Set up the Python packages where needed. In particular, reduce the logging level for some of the used
+    libraries, which are particularly talkative in DEBUG mode. Usually when running in DEBUG mode, we want
+    diagnostics about the model building itself, but not for the underlying libraries.
+    It also adds workarounds for known issues in some packages.
+    """
+    # Numba code generation is extremely talkative in DEBUG mode, disable that.
+    logging.getLogger('numba').setLevel(logging.WARNING)
+    # Matplotlib is also very talkative in DEBUG mode, filling half of the log file in a PR build.
+    logging.getLogger('matplotlib').setLevel(logging.INFO)
+    # Urllib3 prints out connection information for each call to write metrics, etc
+    logging.getLogger('urllib3').setLevel(logging.INFO)
+    logging.getLogger('msrest').setLevel(logging.INFO)
+    # AzureML prints too many details about logging metrics
+    logging.getLogger('azureml').setLevel(logging.INFO)
+    # Jupyter notebook report generation
+    logging.getLogger('papermill').setLevel(logging.INFO)
+    logging.getLogger('nbconvert').setLevel(logging.INFO)
+    # This is working around a spurious error message thrown by MKL, see
+    # https://github.com/pytorch/pytorch/issues/37377
+    os.environ['MKL_THREADING_LAYER'] = 'GNU'
+    # Workaround for issues with matplotlib on some X servers, see
+    # https://stackoverflow.com/questions/45993879/matplot-lib-fatal-io-error-25-inappropriate-ioctl-for-device-on-x
+    # -server-loc
+    matplotlib.use('Agg')
+
+
+def create_runner_parser(model_config_class: Any = None) -> argparse.ArgumentParser:
+    """
+    Creates a commandline parser, that understands all necessary arguments for running a script in Azure,
+    plus all arguments for the given class. The class must be a subclass of GenericConfig.
+
+    :param model_config_class: A class that contains the model-specific parameters.
+    :return: An instance of ArgumentParser.
+    """
+    parser = ExperimentConfig.create_argparser()
+    ModelConfigLoader.add_args(parser)
+    if model_config_class is not None:
+        if not issubclass(model_config_class, GenericConfig):
+            raise ValueError(f"The given class must be a subclass of GenericConfig, but got: {model_config_class}")
+        model_config_class.add_args(parser)
+
+    return parser
+
+
+def additional_run_tags(commandline_args: str) -> Dict[str, str]:
+    """
+    Gets the set of tags from the commandline arguments that will be added to the AzureML run as metadata
+
+    :param commandline_args: A string that holds all commandline arguments that were used for the present run.
+    """
+    return {
+        "commandline_args": commandline_args,
+        CROSS_VALIDATION_SPLIT_INDEX_TAG_KEY: "-1",
+    }
+
+
+def create_experiment_name() -> str:
+    """
+    Gets the name of the AzureML experiment. This is taken from the commandline, or from the git branch.
```

Comment:
different strategy here? Maybe use the model name as the default experiment name?